### DISCUSSION TITLE
Qual é a melhor maneira de medir a performance em Javascript?

### DISCUSSION THREAD

**Discussion by renedocket** [frontendbr/forum#1521] [Perguntas e Respostas] [answered]
Olá pessoal, tudo beleza?

Eu gostaria de saber como que é feito a análise de performance em questão de processamento, carregamento do JS. Tem ferramentas que é possível visualizar se um código não está performático ou algo do tipo?

**Comment by vitorbritto** [comment, marked as answer]
Opa! Sim. Recomendo essas para suas análises de performance:

- [Perflink](https://perf.link)
- [JS Bench](https://jsbench.me)
- [Dromaeo](http://dromaeo.com)

**Comment by renedocket** [comment]
Muito Legal, essas ferramentas! Gostei bastante vitorbritto, muito obrigado! 
Eu procurei aqui, mas não achei documentação deles, você sabe se eles tem algo do tipo?

**Comment by vitorbritto** [comment]
@renedocket 

**Dromaeo:** http://wiki.mozilla.org/Dromaeo
**JS Brench:** https://github.com/psiho/jsbench-me/wiki

Já no _Perflink_, não temos uma documentação. Pelo menos, não conheço uma. 

Porém, é tranquilo. Você posiciona o código que deseja testar e adiciona _cases_ para ter o resultado de cada um e comparação entre eles.

**Comment by eliseumds** [comment]
Primeira e mais essencial pergunta: você tá enfrentando problemas de performance?

Se estiver falando de web development (ou mobile), mais importante é medir *perceived performance* do que focar em trechos específicos de código (a não ser que você esteja lidando com coisa muito grande). Ferramentas como [web.dev](https://web.dev/measure) podem ajudar (e a extensão [Lighthouse](https://chrome.google.com/webstore/detail/lighthouse/blipmdconlkpinefehnmjammfjpmpbjk?hl=en)).
